jupyterhub/oauth/provider.py

```diff
@@ -668,6 +668,15 @@ def add_client(
         self.db.commit()
         return orm_client
 
+    def remove_client(self, client_id):
+        """Remove a client by its id if it is existed."""
+        orm_client = (
+            self.db.query(orm.OAuthClient).filter_by(identifier=client_id).one_or_none()
+        )
+        if orm_client is not None:
+            self.db.delete(orm_client)
```

Comment:
Add an `app_log.info` for deleting the client and `app_log.warning` if there was no client to delete.